From 7f88aa2ef5c42bd58c0a2d9e72c0913491fc16bc Mon Sep 17 00:00:00 2001
From: "Doug.Smith" <doug.smith@lairdconnect.com>
Date: Mon, 4 Jul 2022 12:02:13 -0400
Subject: [PATCH] DHE: remove duplicate public key check

Expensive DHE public key check is being done twice.
Remove EVP_PKEY_public_check() in
tls_process_ske_dhe()<ssl/statem/statem_clnt.c>.
DHE public key check will be done when
EVP_PKEY_derive_set_peer() is called by ssl_derive()<ssl/s3_lib.c>.
---
 ssl/statem/statem_clnt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ssl/statem/statem_clnt.c b/ssl/statem/statem_clnt.c
index b59eddae33..21aa738c8a 100644
--- a/ssl/statem/statem_clnt.c
+++ b/ssl/statem/statem_clnt.c
@@ -2079,7 +2079,7 @@ static int tls_process_ske_dhe(SSL *s, PACKET *pkt, EVP_PKEY **pkey)
              * EVP_PKEY_param_check_quick() skips the safe prime check.
              */
             || EVP_PKEY_param_check_quick(pctx) != 1
-            || EVP_PKEY_public_check(pctx) != 1) {
+        ) {
         SSLfatal(s, SSL_AD_ILLEGAL_PARAMETER, SSL_R_BAD_DH_VALUE);
         goto err;
     }
-- 
2.17.1

